compiler/ballerina-lang/src/main/java/io/ballerina/projects/UsedBIRNodeAnalyzer.java

```diff
@@ -0,0 +1,738 @@
+            new CompilerContext.Key<>();
+    private static final Set<String> USED_FUNCTION_NAMES =
+            new HashSet<>(Arrays.asList("main", ".<init>", ".<start>", ".<stop>", "__execute__"));
+    private static final Map<String, Set<String>> INTEROP_DEPENDENCIES = new HashMap<>();
+    private static final Set<InstructionKind> ANALYZED_INSTRUCTION_KINDS = new HashSet<>(
+            Arrays.asList(InstructionKind.NEW_TYPEDESC, InstructionKind.NEW_INSTANCE, InstructionKind.TYPE_CAST,
+                    InstructionKind.FP_LOAD, InstructionKind.TYPE_TEST, InstructionKind.RECORD_DEFAULT_FP_LOAD,
+                    InstructionKind.NEW_TABLE, InstructionKind.NEW_ARRAY, InstructionKind.MOVE,
+                    InstructionKind.NEW_ERROR));
+    private static final Set<InstructionKind> ANALYZED_TERMINATOR_KINDS =
+            new HashSet<>(Arrays.asList(InstructionKind.CALL, InstructionKind.FP_CALL));
+    private static final String EXTERNAL_METHOD_ANNOTATION_TAG = "Method";
+    private static final String EXECUTE_TEST_REGISTRAR = "executeTestRegistrar";
+    // To check whether a given FP is "USED" or not, we have to keep track of the existing FPs of a given scope.
+    // Variable Declarations holding an FPs are needed to be tracked to achieve that.
+    private Map<BIRNode.BIRVariableDcl, FunctionPointerData> localFpHolders = new HashMap<>();
+    // pkgWiseInvocationData is used for debugging purposes
+    public final Map<PackageID, UsedBIRNodeAnalyzer.InvocationData> pkgWiseInvocationData = new LinkedHashMap<>();
+    protected UsedBIRNodeAnalyzer.InvocationData currentInvocationData;
+    protected boolean isTestablePkgAnalysis = false;
+    protected PackageID currentPkgID;
+    private final PackageCache pkgCache;
+    private final UsedTypeDefAnalyzer usedTypeDefAnalyzer;
+    private List<BIRNonTerminator> currentInstructionArr;
+    private BIRNode.BIRFunction currentParentFunction;
+
+    private UsedBIRNodeAnalyzer(CompilerContext context) {
+        context.put(USED_BIR_NODE_ANALYZER_KEY, this);
+        this.pkgCache = PackageCache.getInstance(context);
+        this.usedTypeDefAnalyzer = UsedTypeDefAnalyzer.getInstance(context);
+        initInteropDependencies();
+    }
+
+    public static UsedBIRNodeAnalyzer getInstance(CompilerContext context) {
+        UsedBIRNodeAnalyzer deadBIRNodeAnalyzer = context.get(USED_BIR_NODE_ANALYZER_KEY);
+        if (deadBIRNodeAnalyzer == null) {
+            deadBIRNodeAnalyzer = new UsedBIRNodeAnalyzer(context);
+        }
+        return deadBIRNodeAnalyzer;
+    }
+
+    public void analyze(BLangPackage pkgNode) {
+        CodeGenOptimizationReportEmitter.flipBirOptimizationTimer(pkgNode.packageID);
+
+        currentInvocationData = pkgNode.symbol.invocationData;
+        pkgWiseInvocationData.putIfAbsent(pkgNode.packageID, currentInvocationData);
+        currentPkgID = pkgNode.packageID;
```

Comment:
Define this at the top of the function and use the variable in the appropriate places

Reply:
Fixed in 9996ad2bea285c54defe7d979395a3d868532b89.